kart/spatial_filters.py

```diff
@@ -37,34 +45,46 @@ def from_spec(cls, geometry_wkt, crs_spec):
 
         return SpatialFilter(filter_geometry_ogr, crs)
 
-    def __init__(self, filter_geometry_ogr, crs, match_all=False):
+    def __init__(
+        self, filter_geometry_ogr, crs, geom_column_name=None, match_all=False
+    ):
         """
         Create a new spatial filter.
         filter_geometry_ogr - The shape of the spatial filter. An OGR Geometry object.
         crs - The CRS used to interpret the spatial filter. An OGR SpatialReference object.
         match_all - if True, this filter is the default match-everything filter.
         """
-        self.filter_ogr = filter_geometry_ogr
-        if self.filter_ogr is not None:
-            self.filter_env = self.filter_ogr.GetEnvelope()
-        else:
+        if match_all:
+            self.filter_ogr = None
             self.filter_env = None
+        else:
+            self.filter_ogr = filter_geometry_ogr
+            self.filter_env = self.filter_ogr.GetEnvelope()
 
         self.crs = crs
+        self.geom_column_name = geom_column_name
         self.match_all = match_all
 
-    def matches(self, feature_geometry):
+    def matches(self, feature_or_geometry):
         """
         Returns True if the given feature geometry matches this spatial filter.
         The feature to be tested is assumed to be using the same CRS as this spatial filter,
         otherwise the intersection test makes no sense.
         To get a spatial filter for a particular CRS, see transfrom_for_dataset / transform_for_crs.
 
-        feature_geometry - a geometry.Geometry object.
+        feature_geometry - either a feature dict (in which case self.geom_column_name must be set)
+            or a geometry.Geometry object.
         """
-        if self.match_all or self.filter_ogr is None or feature_geometry is None:
+        if self.match_all or feature_or_geometry is None:
             return True
 
+        if isinstance(feature_or_geometry, dict):
+            feature_geometry = feature_or_geometry[self.geom_column_name]
+        elif isinstance(feature_or_geometry, Geometry):
```

Comment:
is this used? the two calls I see only pass features in. In which case this function might as well only take features

Reply:
Good point, this no longer receives mere geometries. Removed this capability.